File: MultiLanguageSandbox/src/test/deps/julia/45.jl
function max_product_partition(n::Int, m::Int) :: Vector{Int}
    """
    Find `m` positive integers whose sum is `n`, and their product is as large as possible. Return the partition in lexicographically smallest order.

    # Examples
    - `max_product_partition(6, 3)` should return `[2, 2, 2]` because 2*2*2 is the largest product you can get from three numbers that add up to 6.
    - `max_product_partition(8, 3)` should return `[2, 3, 3]` since 2*3*3 = 18 is the largest product from three numbers that add up to 8.

    # Arguments
    - `n`: The sum of the numbers.
    - `m`: The number of positive integers.

    # Returns
    - A vector of integers representing the partition.
    """
    if m <= 0 || n < m
        error("Invalid input: m must be positive and n must be at least m")
    end
    
    base = n ÷ m
    remainder = n % m
    
    # Create a vector with (m - remainder) elements of base and remainder elements of base + 1
    partition = vcat(fill(base, m - remainder), fill(base + 1, remainder))
    
    # Sort to ensure lexicographical order (though it's already sorted in this approach)
    sort!(partition)
    
    return partition
end
@assert max_product_partition(6, 3) == [2, 2, 2]
@assert max_product_partition(8, 3) == [2, 3, 3]
@assert max_product_partition(10, 2) == [5, 5]
@assert max_product_partition(7, 3) == [2, 2, 3]

File: MultiLanguageSandbox/src/test/deps/C#/45.cs
using System;
using System.Collections.Generic;
using System;
using System.Collections.Generic;
using System.Diagnostics;

class Program
{

/* Finds M positive integers that sum up to N and have the maximum possible product. If multiple solutions exist, 
   returns the one with the lexicographically smallest sequence.
    >>> MaxProductPartition(6, 3)
    [2, 2, 2]

*/


    public static List<int> MaxProductPartition(int N, int M)
{
        List<int> result = new List<int>();
        int baseValue = N / M;
        int remainder = N % M;
        
        for (int i = 0; i < M; i++) {
            if (i < remainder) {
                result.Add(baseValue + 1);
            } else {
                result.Add(baseValue);
            }
        }
        
        // To ensure lexicographical order, we need the smallest numbers first.
        // The above approach already does this because we add the larger numbers first (if any), but then the loop starts from 0 to M-1,
        // so the larger numbers (baseValue +1) are added first if remainder >0. But we need them last for lex smallest.
        // So we need to reverse the logic: the first 'remainder' elements are baseValue+1, the rest are baseValue.
        // But to make it lex smallest, we should have the smallest numbers first. So the correct approach is to have the baseValue numbers first, followed by baseValue+1.
        // So the initial approach is incorrect for lex order. Let's fix it.
        
        // Correct approach: the lex smallest sequence has as many baseValue as possible first, then baseValue+1.
        result.Clear();
        for (int i = 0; i < M; i++) {
            if (i >= M - remainder) {
                result.Add(baseValue + 1);
            } else {
                result.Add(baseValue);
            }
        }
        
        return result;
    }
    static void Main()
    {

        Debug.Assert(MaxProductPartition(6, 3).SequenceEqual(new List<int> { 2, 2, 2 }));
        Debug.Assert(MaxProductPartition(8, 3).SequenceEqual(new List<int> { 2, 3, 3 }));
        Debug.Assert(MaxProductPartition(10, 2).SequenceEqual(new List<int> { 5, 5 }));
        Debug.Assert(MaxProductPartition(7, 3).SequenceEqual(new List<int> { 2, 2, 3 }));

    }
}
